polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -630,6 +630,13 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *expr) {
     return val;
   };
 
+  switch (expr->getBuiltinCallee()) {
```

Comment:
I prefer this code to be placed in `EmitBuiltinOps` were other builtin operations are handled. 
Is possible that the `EmitBuiltinOps` need improvements (I see a lot of string comparisons in that function, would be nice to avoid the string comparisons if possible).

Reply:
https://github.com/intel/llvm/pull/7257